cpp/src/arrow/flight/sql/odbc/entry_points.cc

```diff
@@ -293,5 +313,9 @@ SQLRETURN SQL_API SQLTables(SQLHSTMT stmt, SQLWCHAR* catalogName,
       stmt, fmt::ptr(catalogName), catalogNameLength, fmt::ptr(schemaName),
       schemaNameLength, fmt::ptr(tableName), tableNameLength, fmt::ptr(tableType),
       tableTypeLength);
-  return SQL_ERROR;
+  return ODBC::ODBCStatement::ExecuteWithDiagnostics(stmt, SQL_ERROR, [=]() {
+    throw driver::odbcabstraction::DriverException("SQLTables is not implemented",
+                                                   "IM001");
+    return SQL_ERROR;
+  });
 }
```

Comment:
This block is for local testing only, since Power BI is erroring out on unsupported SQLTables function, will rebase onto `apache-odbc` after #65 is merged and remove this change.